servers/src/main/java/tachyon/worker/block/BlockMetadataManagerView.java

```diff
@@ -107,7 +119,24 @@ public boolean isBlockLocked(long blockId) {
    * @return boolean, true if the block can be evicted
    */
   public boolean isBlockEvictable(long blockId) {
-    return (!isBlockPinned(blockId) && !isBlockLocked(blockId));
+    return (!isBlockPinned(blockId) && !isBlockLocked(blockId) && !isMarkedToMoveOut(blockId));
+  }
+
+  /**
+   * Test if the block is marked to move out of its current dir in this view.
+   *
+   * @param blockId the Id of the block
+   * @return boolean, true if the block is marked to move out
+   */
+  public boolean isMarkedToMoveOut(long blockId) {
```

Comment:
NIT: I think we can simply call this method `isBlockMarked` to keep naming consistent with other methods in this class, and in javadoc to detail this mark means we indent to move it out of the dir.